app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  authorize_resource :topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
Because sometimes the topic_id is a slug. But I think if you use discussion_topic instead of concrete, this can be removed.